packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -166,19 +182,50 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
                         </SimpleFormIteratorItem>
                     ))}
                 </ul>
-                {!disabled && !disableAdd && (
-                    <div className={SimpleFormIteratorClasses.add}>
-                        {cloneElement(addButton, {
-                            className: clsx(
-                                'button-add',
-                                `button-add-${source}`
-                            ),
-                            onClick: handleAddButtonClick(
-                                addButton.props.onClick
-                            ),
-                        })}
-                    </div>
-                )}
+                <div className={SimpleFormIteratorClasses.buttons}>
+                    {!disabled && (
```

Comment:
But then if both `disableAdd` and `disableRemove` are true, we will insert an empty div for nothing...

Reply:
empty div should not render, I agree

resolve commit : 264675654aa539b0bd51061935a680b0faa6127c

While doing this, I realized that array clear button has `add` suffix on classname. I guess the each button's class name should have relate to the its behavior.
I added `poc` commit for verification : dd187c5b5c3bfabceacb995f035db110a9cf4f18